neos/neos-ui
# BUGFIX: Make Guest Frame and Inspector more resilient when there's no focusable document

**What I did**

I prevented the UI from crashing when an account without respective privileges navigates to a document that is matched by a `NodeTreePrivilege`.

**How I did it**

I added several null-checks in places that caused runtime errors when a `NodeTreePrivilege` prevented access to augmented document metadata.

**How to verify it**

1. You'll need a configured NodeTreePrivilege that hides at least one node within the document tree. An example would be:

```yaml
privilegeTargets:
  'Neos\Neos\Security\Authorization\Privilege\NodeTreePrivilege':
    'Vendor.Site:MyNodeTreePrivilege':
      matcher: 'nodeIsOfType("Vendor.Site:Document.Privileged")'

roles:
  'Neos.Neos:Administrator':
    privileges:
      -
        privilegeTarget: 'Vendor.Site:MyNodeTreePrivilege'
        permission: GRANT
```
(This would hide all nodes of type `Vendor.Site:Document.Privileged`)

2. Try to reach the restricted document with an account that doesn't have the above privilege by:
  - Linking to the document, or
  - Entering the backend-URI of the document directly

**Before the fix** you would have seen a blank screen with a console error:
![127741443-21619a14-6fc1-4740-b14b-0292f721217a](https://user-images.githubusercontent.com/2522299/127768989-a592632e-c6e1-4f46-ab37-4d3e1d872a9d.png)

**After the fix** you should see the document without editable content and with an empty inspector.

fixes: #1717,  #2172
